crates/task-impls/src/consensus.rs

```diff
@@ -513,31 +474,27 @@ where
 
             // Poll the future leader for lookahead
             let lookahead_view = new_view + LOOK_AHEAD;
-            if !self.quorum_exchange.is_leader(lookahead_view) {
-                self.quorum_exchange
-                    .network()
+            if self.quorum_membership.get_leader(lookahead_view) != self.public_key {
```

Comment:
I feel like it might be nice to make an "is_leader" function on the membership type, since we will call this same logic often.  That can be for a later PR, though.